src/Processors/QtiV2/In/Validation/BaseInteractionValidationBuilder.php

```diff
@@ -300,6 +301,22 @@ protected function processConditionBranch(QtiComponent $conditionBranch)
                 $results['scoring_type'] = 'match';
                 break;
 
+            case $expression instanceof StringMatch:
+                $responseRules = $conditionBranch->getResponseRules();
+                $caseSensitive = $expression->isCaseSensitive();
+                $subExpressions = $expression->getExpressions();
+
+                // here we expects subExpressions[0] to be a Variable and [1] to be a BaseValue
+                $responseId = $subExpressions[0]->getIdentifier();
```

Comment:
We've had cases where it's not in this order, can we maybe be a bit more defensive/permissive here? Either way we should check whether `$subExpressions[0]` is an "identifiable" entity first.